src/Compilers/Core/Desktop/AnalyzerFileReference.InMemoryAssemblyLoader.cs

```diff
@@ -193,6 +255,55 @@ private static Assembly CurrentDomain_AssemblyResolve(object sender, ResolveEven
                     return assembly;
                 }
             }
+
+            private static AssemblyIdentity TryGetAssemblyIdentity(string filePath)
+            {
+                try
+                {
+                    using (var stream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read | FileShare.Delete))
```

Comment:
Probably worth adding a check for File.Exists before trying to open it here? Seems like we will call it for all combinations of "<LoadedAssemblyDir>.RequestedAssemblyName", lot of which are non-existent paths...